CameraInfo/Program.cs

```diff
@@ -43,7 +40,25 @@ static void Main(string[] args)
             Result<string> deviceNameResult = await lastDevice.GetName();
             string deviceName = deviceNameResult.IsSuccess ? deviceNameResult.Value : "Unknown";
 
-            Console.WriteLine($"Device type {deviceModel} with serial number {serialNumber} and name {deviceName} is manufactured by {manufacturer}.");
+            Console.WriteLine(
+                $"Device type {deviceModel} with serial number {serialNumber} and name {deviceName} is manufactured by {manufacturer}."
+            );
+
+            // Device firmware version
+            if (lastDevice.Model != DeviceModel.UsbAdapter)
+            {
+                var fwVersion = (await lastDevice.GetFwVersion()).Value;
+                Console.WriteLine($"Device firmware version: {fwVersion?.ToString() ?? "unknown"}");
+            }
+            // USB adapter firmware version
+            if (lastDevice is UsbAdapterDevice usbAdapterDevice)
+            {
+                var usbAdapterFwVersion = (await usbAdapterDevice.GetUsbAdapterFwVersion()).Value;
+                Console.WriteLine(
+                    $"USB adapter firmware version: {usbAdapterFwVersion?.ToString() ?? "unknown"}"
+                );
+            }
```

Comment:
Not  sure about  addig this. It complicates the  example with very specific logic that consumers ideally should never  have to  think about. Better to communicate how to check  the adapter version somewhere else.
@pkhansen thoughts?

Reply:
Yes it's something to consider. I don't really like putting all kinds of very specific calls in the IDevice interface though. For now maybe consumers can just stick to doing what you've done here. I just don't think that the example really _needs_ to show it. Consumers should only care about the adapter version in very specific use cases, or, that's what I hope for at least.

Reply:
I'm a bit groggy when it comes to details in the dotnet-sdk, but the intent is to show the firmware version of the smartbase device if the device is connected through a smartbase?

I agree that it is always better to keeps these samples focused and to the point, but I'm not sure where this would belong.